src/platforms/posix/drivers/df_lsm9ds1_wrapper/df_lsm9ds1_wrapper.cpp

```diff
@@ -671,13 +665,13 @@ int DfLsm9ds1Wrapper::_publish(struct imu_sensor_data &data)
 	accel_report.z = accel_val_filt(2);
 
 	if (_mag_enabled) {
-
+		// the sensor coordinate system of mag is different. MAG_X should be -MAG_X
+		data.mag_ga_x = - data.mag_ga_x;
```

Comment:
This should be handled in the driver itself.

I see swapping of the z-axis here:
https://github.com/PX4/DriverFramework/blob/master/drivers/lsm9ds1/LSM9DS1.cpp#L403-L417

Are you sure the mag x needs an additional swap? If yes, please make a PR for the DriverFramework.